### ISSUE TITLE
Adding pytest's `filterwarnings` causes 0% coverage

### ISSUE THREAD

**Issue by g-votte** [optuna/optuna#2517]
## Steps to reproduce

1. Apply the change of https://github.com/optuna/optuna/pull/2405 , which adds `filterwarnings` of pytest.
2. Run pytest with coverage reporting

## Current behavior

The coverage drastically downs and shows [0% in the CI](https://app.codecov.io/gh/optuna/optuna/compare/2405/overview).

## Expected behavior

The coverage does not change.

## Reproducible examples

You can quickly reproduce the drastic decrease by locally running the following command and checking the output report.

```console
pytest --cov=optuna --cov-report=html tests/study_tests/test_study.py
```

## Additional context (optional)

- https://github.com/optuna/optuna/pull/2405 was reverted in https://github.com/optuna/optuna/pull/2413 due to the coverage decrease.
- We want to re-apply the `filterwarning` after resolving this issue, which removes a lot of noises in the unit tests' log.
- This issue also blocks https://github.com/optuna/optuna/pull/2416.
- This issue was also investigated in https://github.com/optuna/optuna/pull/2417 . However, adding parentheses did not resolve the problem.

**Comment by g-votte** [comment]
Some more attempts have been made in #2529. However, neither changing the `pytest` version nor adding the `--cov-append` option did not solve the problem.

**Comment by jeromepatel** [comment]
Actually, I tried to solve the issue by trying different things but none of them did work. Here's the summary of what I have tried:
* Changed configuration to `tox.ini` 
* Added missing `__init.py__` file inside a test subdirectory 
* Removed coverage for `mpirun` (multi-node test)
* Changed `pytest` version and `pytest-cov` version (separately )  </br>      
Unfortunately, none of them resolved the current issue. I apologize for that.
